### PR TITLE
[TRTLLM-15405][refactor] BREAKING: Remove TRTLLMSampler and sampler_type

### PR DESCRIPTION
**Pull request by zhaoyangwang-nvidia**
## Dev Engineer Review

- Removes `TRTLLMSampler`, `SampleState` types, `SamplerType`, and `TorchLlmArgs.sampler_type`.
- Makes `TorchSampler` the only regular sampling backend.
- Updates AutoDeploy and executor paths to construct `TorchSampler` directly.
- Removes sampler-specific validation, synchronization, helpers, documentation, and configuration entries.
- Leaves the C++ decoder stack unchanged and unreachable from the Python path.
- Updates API references, schemas, telemetry, examples, pre-commit lists, and Ruff lists.
- Updates beam-search logprob and speculative-decoding metric handling.
- Removes obsolete TRTLLM sampler entries from test lists and related duration data.
- The second CI run succeeded, but its associated pipeline was unstable. Multi-GPU tests require the `ci: full pre-merge approved` label.

## QA Engineer Review

Modified or removed coverage includes:

- Removed `test_trtllm_sampler`.
- Removed `test_trtllm_sampler_with_stop_token_ids`.
- Removed `test_torch_sampler_with_multi_token_stop_words`.
- Removed `test_trtllm_sampler_best_of_with_logprobs`.
- Removed `test_ad_trtllm_sampler_smoke`.
- Updated beam-search validation, `TorchSampler`, overlap-scheduler, penalty, speculative-decoding, logprob, API stability, and telemetry tests.
- Removed sampler-specific test parameterization.
- Updated logit-bias chat and completion tests.

The logit-bias tests remain covered by `tests/integration/test_lists/test-db/l0_a30.yml`. The removed TRTLLM sampler tests were removed from the A10 list and related duration and standalone-package mappings. Other modified unit tests have no clear dedicated `test-db/` or `qa/` entries.

**Verdict: needs follow-up.** The second CI run succeeded but was unstable. Add the required pre-merge label and rerun CI. Complete CI coverage mapping is unavailable.

## Description

JIRA: [TRTLLM-15405](https://jirasw.nvidia.com/browse/TRTLLM-15405)

## Removal plan

`TRTLLMSampler` is removed in two steps so that every user-visible change lands
in one small review and the bulk deletion stays a pure dead-code change:

| Step | Scope | Size |
|---|---|---|
| **1** | Python sampler + the `sampler_type` LLM API option, plus tests and docs | 47 files, −1391 |
| **2** | The C++ decoder stack that step 1 leaves unreachable | 131 files, −34126 |

Step 1 carries all of the API risk and is independently correct: afterwards the
C++ decoder still builds, it simply has no caller. Step 2 then deletes it
without touching behaviour.

Step 2 is not split further because the subsystem is mutually recursive —
`layers/` includes `runtime/decodingLayerWorkspace.h` while
`runtime/gptDecoder.cpp` and `decodingLayerWorkspace.h` include
`layers/decodingParams.h`, so deleting either side alone leaves the other
uncompilable.

**This PR is step 1.**

`TorchSampler` becomes the only sampling backend. This has been signposted for a while: `sampler_type` already carries a `status="deprecated"` marker and a *"will be removed in release 1.4"* notice, and `TorchSampler` has supported a superset of `TRTLLMSampler`'s features for several releases (the docs stated as much). `auto` already resolved to `TorchSampler`, so the default path is unchanged.

Removed: the `TRTLLMSampler` class and its `SampleState*` types, `make_decoding_batch_input_output.py`, and the helpers only it reached (`get_decoding_mode()`, `update_sampler_max_seq_len()`), plus the now-dead `trtllm_sampler` feature-conflict rules.

**API change.** The `sampler_type` option and its `SamplerType` enum are removed outright. Once `TRTLLMSampler` is gone, `auto` and `TorchSampler` resolve to the same sampler, so the option no longer selects anything — keeping a single-valued knob would only be a way to get it wrong. `auto` was the default, so code that never set it is unaffected; code that passed `sampler_type=` (including `sampler_type:` in a YAML `--extra_llm_api_options` file) must drop the line. Both consumers are simplified in step: the AutoDeploy shim constructs `TorchSampler` directly instead of branching, and `validate_feature_combination()` drops its `sampler_type` parameter along with the `torch_sampler` feature flag, which fed no conflict rule. The telemetry allowlist, golden manifest, API-stability reference and docs are updated to match.

This PR touches no C++ at all, so the C++ build is unaffected; the decoder stack it orphans is deleted in step 2.

## Test Coverage

Existing tests. Coverage specific to the removed sampler is deleted; everything else is re-pointed at `TorchSampler`:

- Deleted: `tests/unittest/_torch/sampler/test_trtllm_sampler.py`, `tests/unittest/auto_deploy/singlegpu/smoke/test_ad_trtllm_sampler.py`, `TestParameterValidation::test_logprobs_trtllm_sampler` (asserted a C++-decoder-only error), and `test_llmapi_config_capture.py::test_collect_llm_api_config_captures_sampler_type_categorical` (`load_format` still covers the `Union[str, Enum]` telemetry path).
- Sampler parametrization is dropped entirely (it now has a single value) across `test_beam_search.py`, `test_logits_logprobs.py`, `test_overlap_scheduler.py`, `_test_openai_chat.py`, `_test_openai_completions.py` and `test_e2e.py`. Two guards in `test_beam_search.py` had degenerated with the parameter (one skipped unconditionally, one never skipped) and are removed rather than left misleading. Test ids shift as a result, so the affected `waives.txt` and `test-db/l0_a30.yml` entries are updated, and the `test_e2e.py` wrappers now filter on `-k logit_bias_effect` instead of the removed `-k torch_sampler`, which would otherwise have matched nothing and silently run zero tests.
- `test_beam_search.py::test_beam_search_large_beam_width_regression` (nvbugs/6242591) is kept and switched to `TorchSampler` — it exercises beam-slot reuse, which is not sampler-specific.

Local verification: `pre-commit` passes on all changed files.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: docs/source/features/sampling.md [hunk 1]
```diff
@@ -17,27 +17,8 @@ The PyTorch backend supports a wide variety of features, listed below:
...
-          sampler_type="TorchSampler")
-
-# Chooses TRTLLMSampler explicitly
-llm = LLM(model='nvidia/Llama-3.1-8B-Instruct-FP8',
-          sampler_type="TRTLLMSampler")
-```
-
-By default, the sampling backend is chosen to be `auto`. This will use Torch Sampler for all requests.
+Torch Sampler is the only sampling backend and is used for all requests; there
+is nothing to configure.
```
Comment (Funatiq) [inline docs/source/features/sampling.md]: I think we can even remove this sentence completely.

File: tensorrt_llm/_torch/pyexecutor/sampler/sampler.py
Comment (Funatiq) [inline tensorrt_llm/_torch/pyexecutor/sampler/sampler.py]: I think these can also be removed now.

File: tensorrt_llm/_torch/pyexecutor/py_executor.py [hunk 2]
```diff
@@ -692,15 +692,6 @@ def __init__(
             "TRTLLM_PP_MULTI_STREAM_SAMPLE", "1") == "1"
         self.sample_stream = torch.cuda.Stream()
         self.finish_sample_event = torch.cuda.Event()
-        if (self.dist.pp_size > 1 and self.pp_multi_stream_sample
```
Comment (Funatiq) [inline tensorrt_llm/_torch/pyexecutor/py_executor.py:729]: It seems like `pp_multi_stream_sample` is not used anywhere else. Should we remove it completely?

File: tensorrt_llm/executor/result.py [hunk 3]
```diff
@@ -295,8 +298,8 @@ def _maybe_fill_spec_dec_perf_metrics(
         drafting ran. The PyTorch executor instead attaches cumulative
         (accepted, drafted) totals to the response (LlmResult.spec_dec_totals,
         stashed on self in _handle_response); fill the section from them.
-        No-op when the section is already populated (TRT engine / TRTLLMSampler
-        paths) or when no drafting occurred.
+        No-op when the section is already populated (TRT engine path) or when
+        no drafting occurred.
```
Comment (Funatiq) [inline tensorrt_llm/executor/result.py]: Since TRT engine path does not exist anymore we should remove that comment.

File: tests/integration/defs/.test_durations [hunk 4]
```diff
@@ -989,10 +989,8 @@
    "test_e2e.py::test_openai_chat_harmony_perf_metrics": 161.03756944444444,
    "test_e2e.py::test_openai_chat_multimodal_example": 127.38494212962964,
    "test_e2e.py::test_openai_chat_with_logit_bias[torch_sampler]": 157.82800883002207,
```
Comment (Funatiq) [inline tests/integration/defs/.test_durations]: Remove the `[torch_sampler]` parameter here?

File: tests/integration/defs/.test_durations [hunk 5]
```diff
@@ -989,10 +989,8 @@
...
-   "test_e2e.py::test_openai_chat_with_logit_bias[trtllm_sampler]": 157.81243107221007,
    "test_e2e.py::test_openai_completions_example[pytorch]": 311.7045913242009,
    "test_e2e.py::test_openai_completions_with_logit_bias[torch_sampler]": 70.47549118942732,
```
Comment (Funatiq) [inline tests/integration/defs/.test_durations]: Remove the `[torch_sampler]` parameter here?

File: tests/unittest/executor/test_spec_dec_perf_metrics.py [hunk 6]
```diff
@@ -65,7 +65,7 @@ def test_noop_without_totals():
 
 
 def test_noop_when_section_already_populated():
-    # TRT-engine / TRTLLMSampler paths populate the section runtime-side
+    # The TRT-engine path populates the section runtime-side
     # (updateNumTokensPerIteration); the backfill must not overwrite it.
```
Comment (Funatiq) [inline tests/unittest/executor/test_spec_dec_perf_metrics.py]: Not sure if we still need this test.

File: docs/source/features/sampling.md (hunk 1 above)
Comment (zhaoyangwang-nvidia) [reply docs/source/features/sampling.md]: Done, removed.

File: tensorrt_llm/_torch/pyexecutor/sampler/sampler.py
Comment (zhaoyangwang-nvidia) [reply tensorrt_llm/_torch/pyexecutor/sampler/sampler.py]: Done — also dropped the now-unused `DecoderModel`/`PretrainedConfig` type-only imports.

File: tensorrt_llm/_torch/pyexecutor/py_executor.py (hunk 2 above)
Comment (zhaoyangwang-nvidia) [reply tensorrt_llm/_torch/pyexecutor/py_executor.py:729]: It's still used at `py_executor.py:2743` and `:4885` (and in `test_kv_pool_rebalance.py`) for the TorchSampler PP path — only the TRTLLMSampler branch is removed here.

File: tensorrt_llm/executor/result.py (hunk 3 above)
Comment (zhaoyangwang-nvidia) [reply tensorrt_llm/executor/result.py]: Done.

File: tests/integration/defs/.test_durations (hunk 4 above)
Comment (zhaoyangwang-nvidia) [reply tests/integration/defs/.test_durations]: Done — I'd updated the ids in `waives.txt` and `l0_a30.yml` but missed this file.

File: tests/integration/defs/.test_durations (hunk 5 above)
Comment (zhaoyangwang-nvidia) [reply tests/integration/defs/.test_durations]: Done.

File: tests/unittest/executor/test_spec_dec_perf_metrics.py (hunk 6 above)
Comment (zhaoyangwang-nvidia) [reply tests/unittest/executor/test_spec_dec_perf_metrics.py]: You're right — `updateNumTokensPerIteration` has no callers left, so the branch it guarded is unreachable; removed both the test and the branch.

File: tensorrt_llm/_torch/pyexecutor/py_executor.py (hunk 2 above)
Comment (Funatiq) [reply tensorrt_llm/_torch/pyexecutor/py_executor.py:729]: Thanks, I missed that.

**Review by lori-ren — APPROVED** [review]

**Review by DomBrown — APPROVED** [review]
Approving API changes

**Review by Shixiaowei02 — APPROVED** [review]

**Review by suyoggupta — APPROVED** [review]

**Review by QiJune — APPROVED** [review]
LGTM

**Review by govind-ramnarayan — APPROVED** [review]

**Review by arysef — APPROVED** [review]
